lib/state.js

```diff
@@ -460,7 +460,8 @@ StateResolver.prototype.getMirrorProperties_ = function(mirror) {
 
 StateResolver.prototype.resolveMirrorProperty_ = function(property) {
   var name = String(property.name());
-  if (property.isNative()) {
+  var isArrayLen = property.mirror_.isArray() && name === 'length';
```

Comment:
What is the type of `property`. Is it guaranteed to have a valid `mirror_` property?